src/librustdoc/passes/check_custom_code_classes.rs

```diff
@@ -0,0 +1,89 @@
+//! NIGHTLY & UNSTABLE CHECK: custom_code_classes_in_docs
+//!
+//! This pass will produce errors when finding custom classes outside of
+//! nightly + relevant feature active.
+
+use super::{span_of_attrs, Pass};
+use crate::clean::{Crate, Item};
+use crate::core::DocContext;
+use crate::fold::DocFolder;
+use crate::html::markdown::{find_testable_code, ErrorCodes, LangString};
+
+use rustc_session::parse::feature_err;
+use rustc_span::symbol::sym;
+
+crate const CHECK_CUSTOM_CODE_CLASSES: Pass = Pass {
+    name: "check-custom-code-classes",
+    run: check_custom_code_classes,
+    description: "check for custom code classes while not in nightly",
```

Comment:
```suggestion
    description: "check for custom code classes without a feature-gate enabled",
```

Reply:
Or, since it has to be a specific feature gate:

```suggestion
    description: "check for custom code classes without the feature-gate enabled",
```

Reply:
I'm suddenly wondering if the negative here is correct. Maybe it should be `with the feature-gate enabled` ? I don't remember in which context this message pops up.

Reply:
This doesn't pop up anywhere except `-Z time-passes` I think. Since this pass is explicitly only emitting lints and not doing anything else, I think @camelid's suggestion is fine.

Reply:
@poliorcetics this is still waiting on you to accept the change I think.